agent/api/task/task_test.go

```diff
@@ -5487,3 +5489,310 @@ func TestHasAContainerWithResolvedDigest(t *testing.T) {
+					{
+						Type: apicontainer.ContainerNormal,
+						Name: "client",
+					},
+					{
+						Name: "sc-container",
+					},
+				},
+				ServiceConnectConfig: &serviceconnect.Config{ContainerName: "sc-container"},
+			},
+			instanceIPCompat: ipcompatibility.NewIPCompatibility(true, false),
+			expectedContainerMapping: map[string]string{
+				"APPNET_CONTAINER_IP_MAPPING": `{"client":"8.8.8.8","web":"1.2.3.4"}`,
+			},
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.task.PopulateServiceConnectContainerMappingEnvVarBridge(tc.instanceIPCompat)
+			if tc.expectedError != "" {
+				assert.EqualError(t, err, tc.expectedError)
+			} else {
+				require.NoError(t, err)
+				container := tc.task.GetServiceConnectContainer()
+				assert.Equal(t, tc.expectedContainerMapping, container.Environment)
+			}
+		})
+	}
+}
+
+func TestPopulateServiceConnectContainerMappingEnvVarAwsvpc(t *testing.T) {
+	tcs := []struct {
+		name                     string
+		task                     *Task
+		expectedError            string
+		expectedContainerMapping map[string]string
+	}{
+		{
+			name:          "service connect container not found",
+			task:          &Task{},
+			expectedError: "no primary ENI found in task",
+		},
+		{
+			name: "multiple containers",
+			task: &Task{
+				Containers: []*apicontainer.Container{
+					{
+						Type: apicontainer.ContainerCNIPause,
+					},
+					{
+						Type: apicontainer.ContainerNormal,
+						Name: "web",
+					},
+					{
+						Type: apicontainer.ContainerNormal,
+						Name: "client",
+					},
+					{
+						Name: "sc-container",
+						Type: apicontainer.ContainerCNIPause,
+					},
+				},
+				ServiceConnectConfig: &serviceconnect.Config{ContainerName: "sc-container"},
+				ENIs: []*ni.NetworkInterface{
+					{IPV6Addresses: []*ni.IPV6Address{{Address: "1:2:3:4::"}}},
+				},
+			},
+			expectedContainerMapping: map[string]string{
+				"APPNET_CONTAINER_IP_MAPPING": `{"client":"::1","web":"::1"}`,
+			},
+		},
+		{
+			name: "not ipv6 only",
+			task: &Task{
+				Containers: []*apicontainer.Container{
+					{
+						Type: apicontainer.ContainerCNIPause,
+					},
+					{
+						Type: apicontainer.ContainerNormal,
+						Name: "web",
+					},
+					{
+						Name: "sc-container",
+						Type: apicontainer.ContainerCNIPause,
+					},
+				},
+				ServiceConnectConfig: &serviceconnect.Config{ContainerName: "sc-container"},
+				ENIs: []*ni.NetworkInterface{
+					{
+						IPV4Addresses: []*ni.IPV4Address{{Primary: true, Address: "1.2.3.4"}},
+						IPV6Addresses: []*ni.IPV6Address{{Address: "5:5:5:5::"}},
+					},
+				},
+			},
+			expectedContainerMapping: nil,
+		},
+		{
+			name: "ipv6 only",
+			task: &Task{
+				Containers: []*apicontainer.Container{
+					{
+						Type: apicontainer.ContainerCNIPause,
+					},
+					{
+						Type: apicontainer.ContainerNormal,
+						Name: "web",
+					},
+					{
+						Name: "sc-container",
+						Type: apicontainer.ContainerCNIPause,
```

Comment:
Are sc-container of type `ContainerCNIPause` always?

Reply:
It's actually NORMAL. I will update this and also include the container in IP mapping to make it consistent with bridge case.

Reply:
Updated